crates/sdk/src/masp.rs

```diff
@@ -1777,21 +1751,20 @@ impl<U: ShieldedUtils + MaybeSend + MaybeSync> ShieldedContext<U> {
                             memo.clone(),
                         )
                         .map_err(builder::Error::SaplingBuild)?;
-                } else {
+                } else if let Some(t_addr_data) = target.t_addr_data() {
                     // If there is a transparent output
-                    let hash = transparent_target_hash
-                        .expect(
-                            "transparent target hash should have been \
-                             computed already",
-                        )
-                        .into();
                     builder
                         .add_transparent_output(
-                            &TransparentAddress(hash),
+                            &t_addr_data.taddress(),
                             *asset_type,
                             contr,
                         )
                         .map_err(builder::Error::TransparentBuild)?;
+                } else {
+                    unreachable!(
```

Comment:
Better to return an error from here instead of panicking?

Reply:
Thanks. Resolved.